### PR TITLE
[#172076948] Added a new component for instabug contextual help

### PR DESCRIPTION
**Pull request by Edigio**
Added a new component for instabug contextual help.

AFTER THE INTEGRATION
InstabugInotHelper

### CODE REVIEW

**Comment by Undermaken** [comment]
@valentinacala I guess this is not **wip** anymore. Is it right?

File: locales/en/index.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 global:
+  ioURL: 'io.italia.it'
```
Comment (Undermaken) [inline locales/en/index.yml:3]: It seems to be never used

File: locales/en/index.yml [hunk 2]
```diff
@@ -996,6 +997,15 @@ btnUpdateApp: Update the IO app
...
   closeApp: close the application and try again
+instabug:
+  contextualHelp:
+    title1: Do you need help?
+    buttonChat: Send us a message
+    descriptionChat: You can use our chat tool to send a message in case of urgent need for assistance. You will receive an answer directly on IO
+    buttonBug: Report a bug
+    descriptionBug: You can report an error or something that does not work to our developers, they will take care of the problem as soon as possible.
+    title2: Are you new to IO?
+    descriptionLink: Find out how it works and what are the special features of IO, by visiting the site
```
Comment (Undermaken) [inline locales/en/index.yml:1008]: It seems to be never used

File: locales/en/index.yml [hunk 3]
```diff
@@ -996,6 +997,15 @@ btnUpdateApp: Update the IO app
...
   title: Unable to start IO
   closeApp: close the application and try again
+instabug:
+  contextualHelp:
+    title1: Do you need help?
+    buttonChat: Send us a message
+    descriptionChat: You can use our chat tool to send a message in case of urgent need for assistance. You will receive an answer directly on IO
+    buttonBug: Report a bug
+    descriptionBug: You can report an error or something that does not work to our developers, they will take care of the problem as soon as possible.
+    title2: Are you new to IO?
```
Comment (Undermaken) [inline locales/en/index.yml:1007]: It seems to be never used

File: ts/components/ButtonWithImage.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  light?: boolean;
+};
+
+// Create a button consisting of an icon + label
+class ButtonWithImage extends React.PureComponent<Props> {
+  public render() {
+    const { icon, onClick, text, disabled, light } = this.props;
+    return (
+      <ButtonDefaultOpacity
+        disabled={disabled !== undefined ? disabled : false}
```
Comment (Undermaken) [inline ts/components/ButtonWithImage.tsx:98]:
```suggestion
        disabled={disabled === true}
```

File: ts/components/InstabugAssistanceComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+type Props = ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps> &
+  OwnProps;
+
+type State = {
+  instabugReportType: Option<string>;
+};
+
+// estimated. Values below 250 makes modal being partially displayed on the screen shot
```
Comment (Undermaken) [inline ts/components/InstabugAssistanceComponent.tsx:30]:
```suggestion
// estimated. Values below 250 makes modal being partially displayed on the screenshot
```

File: ts/utils/deepLink.ts [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import { Platform } from "react-native";
 import { NavigationNavigateActionPayload } from "react-navigation";
 
+export const ioItaliaLink = "https://io.italia.it/";
```
Comment (Undermaken) [inline ts/utils/deepLink.ts:4]: It seems to be never used

File: ts/components/InstabugAssistanceComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  instabugReportOpened
+} from "../store/actions/debug";
+import { updateInstabugUnreadMessages } from "../store/actions/instabug";
+import { Dispatch } from "../store/actions/types";
+import { instabugMessageStateSelector } from "../store/reducers/instabug/instabugUnreadMessages";
+import { GlobalState } from "../store/reducers/types";
+import ButtonWithImage from "./ButtonWithImage";
+
+type OwnProps = Readonly<{
+  hideComponent: () => void;
```
Comment (Undermaken) [inline ts/components/InstabugAssistanceComponent.tsx:19]:
```suggestion
  onReportOpening: () => void;
```
Could it be a better name?
BTW we can add another prop `onReportOpened` fired when we open the Instabug report (if we will adopt this solution based on a timer)

**Comment by valentinacala** [comment]
@Undermaken it is wip because it works on the usage of a timeout. I'm working on another approach, as you can see here: https://github.com/pagopa/io-app/pull/1677. It works but I'm managing a formal problem on type definitions (as you can see, the BaseScreenComponent with HOC misses the component props definitions )

File: locales/en/index.yml (hunk 2 above)
Comment (valentinacala) [reply locales/en/index.yml:1008]: it is used in the new component `InstabugAssistanceComponent`

File: ts/components/InstabugAssistanceComponent.tsx (hunk 7 above)
Comment (valentinacala) [reply ts/components/InstabugAssistanceComponent.tsx]: there is an existing fired action, `instabugReportOpened` so I opted for `onInstabugReportOpening`. updated within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

File: ts/utils/deepLink.ts (hunk 6 above)
Comment (valentinacala) [reply ts/utils/deepLink.ts]: removed within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

File: ts/components/InstabugAssistanceComponent.tsx (hunk 5 above)
Comment (valentinacala) [reply ts/components/InstabugAssistanceComponent.tsx]: updated within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

File: ts/components/ButtonWithImage.tsx (hunk 4 above)
Comment (valentinacala) [reply ts/components/ButtonWithImage.tsx]: updated within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

File: locales/en/index.yml (hunk 3 above)
Comment (valentinacala) [reply locales/en/index.yml]: unused strings removed within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

File: locales/en/index.yml (hunk 2 above)
Comment (valentinacala) [reply locales/en/index.yml]: unused strings removed within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

File: locales/en/index.yml (hunk 1 above)
Comment (valentinacala) [reply locales/en/index.yml]: removed within https://github.com/pagopa/io-app/pull/1629/commits/da1bbdfd04c40079d66e00455e3705a3b9a5b0c9

**Comment by valentinacala** [comment]
@Undermaken having some type-related problems on the solution proposed with https://github.com/pagopa/io-app/pull/1677 (it works as expected on both iOs and Android but there is something going wrong with the redux `connect` hoc usage) I propose to integrate this pr and consider the https://github.com/pagopa/io-app/pull/1677 as a further possible optimization. 

https://github.com/pagopa/io-app/pull/1629/commits/405df016dcfea6c2e04544a0d436dddd8d47b189 integrates here some updates (eg the email is no more requested in the chat form)

**Pull request merged**